src/core/src/pass/visualize_tree.cpp

```diff
@@ -505,6 +505,44 @@ string pass::VisualizeTree::get_node_name(shared_ptr<Node> node) {
     }
     rc += "\\n" + (nvtmn ? string("type_name: ") : "") + std::string(node->get_type_name());
 
+    static const bool nvttn = getenv_bool("OV_VISUALIZE_TREE_TENSORS_NAME");
+    if (nvttn) {
+        auto to_string = [](const std::unordered_set<std::string>& names) {
+            std::stringstream ss;
+            size_t i = 0;
+            for (const auto name : names) {
+                ss << (i == 0 ? "" : ", ") << name;
+                i++;
+            }
+            return ss.str();
+        };
+
+        if (node->get_input_size() != 0) {
+            rc += "\\n" + (nvtmn ? string("in_tensor_names: ") : "");
```

Comment:
I believe it's better to use `stringstream` instead of `string` for `rc` variable